### PR TITLE
[19354] Add XML parser bit_bound bounds check

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

Bit mask's bit bound attribute can be set to values out of expected bounds (0 < bit_bound < 65) through XML configuration. This PR add a check in XML parser to avoid it.

@Mergifyio backport 2.11.x 2.10.x 2.6.x

## Contributor Checklist
- [x] Commit messages follow the project guidelines. 
- [x] The code follows the style guidelines of this project. 
- [x] Tests that thoroughly check the new feature have been added/Regression tests checking the bug and its fix have been added; the added tests pass locally 
- **N/A** Any new/modified methods have been properly documented using Doxygen. 
- [x] Changes are ABI compatible. 
- [x] Changes are API compatible. 
- **N/A**  New feature has been added to the `versions.md` file (if applicable).
- **N/A**  New feature has been documented/Current behavior is correctly described in the documentation. 
    
- [x] Applicable backports have been included in the description.

## Reviewer Checklist
- [x] The PR has a milestone assigned.
- [x] Check contributor checklist is correct.
- [x] Check CI results: changes do not issue any warning.
- [x] Check CI results: failing tests are unrelated with the changes.

### CODE REVIEW

**Comment by JesusPoderoso** [comment]
@richiprosima please test mac please test windows

File: src/cpp/rtps/xmlparser/XMLDynamicParser.cpp [hunk 1]
```diff
@@ -619,8 +619,13 @@ XMLP_ret XMLParser::parseXMLBitmaskDynamicType(
         bit_bound = static_cast<uint16_t>(std::atoi(anno_bit_bound));
     }
 
+    if (bit_bound < 1 || bit_bound > 64)
+    {
+        return XMLP_ret::XML_ERROR;
+    }
```
Comment (MiguelCompany) [inline src/cpp/rtps/xmlparser/XMLDynamicParser.cpp:625]:
```suggestion
        auto input_bit_bound = std::atoi(anno_bit_bound);
        if (input_bit_bound < 1 || input_bit_bound > 64)
        {
            return XMLP_ret::XML_ERROR;
        }
        bit_bound = static_cast<uint16_t>(input_bit_bound);
    }
```

**Review by MiguelCompany — CHANGES_REQUESTED** [review]

**Review by MiguelCompany — APPROVED** [review]

**Comment by MiguelCompany** [comment]
@richiprosima Please test this

**Comment by MiguelCompany** [comment]
@richiprosima Please test mac

**Comment by JesusPoderoso** [comment]
@richiprosima please test mac

**Comment by MiguelCompany** [comment]
@Mergifyio backport 2.11.x 2.10.x 2.6.x

**Pull request merged**